pkg/netid/netid.go

```diff
@@ -0,0 +1,93 @@
+package netid
+
+/*
+Accessor methods to annotate namespace for multitenant support
+*/
+
+import (
+	"fmt"
+	"strconv"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+const (
+	// Maximum VXLAN Virtual Network Identifier(VNID) as per RFC#7348
+	MaxVNID = uint((1 << 24) - 1)
+	// VNID: 1 to 9 are internally reserved for any special cases in the future
+	MinVNID = 10
+	// VNID: 0 reserved for default namespace and can reach any network in the cluster
+	GlobalVNID = uint(0)
+
+	// Current assigned VNID for the namespace
+	VNIDAnnotation string = "pod.network.openshift.io/multitenant.vnid"
+	// Desired VNID for the namespace
+	RequestedVNIDAnnotation string = "pod.network.openshift.io/multitenant.requested-vnid"
+)
+
+var (
+	ErrorVNIDNotFound = fmt.Errorf("VNID or RequestedVNID annotation not found")
+)
+
+func ValidVNID(vnid uint) error {
+	if vnid == GlobalVNID {
+		return nil
+	}
+	if vnid < MinVNID {
+		return fmt.Errorf("VNID must be greater than or equal to %d", MinVNID)
+	}
+	if vnid > MaxVNID {
+		return fmt.Errorf("VNID must be less than or equal to %d", MaxVNID)
+	}
+	return nil
+}
+
+func GetVNID(ns *kapi.Namespace) (uint, error) {
+	return getVNIDAnnotation(ns, VNIDAnnotation)
+}
+
+func SetVNID(ns *kapi.Namespace, id uint) error {
+	return setVNIDAnnotation(ns, VNIDAnnotation, id)
+}
+
+func DeleteVNID(ns *kapi.Namespace) {
+	delete(ns.Annotations, VNIDAnnotation)
+}
+
+func GetRequestedVNID(ns *kapi.Namespace) (uint, error) {
```

Comment:
But since you don't have that package it's fine, just add GoDoc.  We generally don't fiddle with this sort of helper wrapper.  Instead, we'd generally expect you to define a `type VNID uint` and have methods that return to and from it.

Again, fine for here.